src/test/scala/chiselTests/naming/ClassNameSpec.scala

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests.naming
+
+import chisel3._
+import chisel3.aop.Select
+import chisel3.experimental.prefix
+import chiselTests.{ChiselFunSpec, Utils}
+import circt.stage.ChiselStage
+
+import scala.collection.immutable.SeqMap
+
+object StaticObject {
+  class MyStaticRecord extends Record { val elements = SeqMap.empty }
+  class MyStaticBundle extends Bundle
+  class MyStaticModule extends Module
+}
+
+class ClassNameSpec extends ChiselFunSpec with Utils {
+  object DynamicObject {
+    class MyDynamicRecord extends Record { val elements = SeqMap.empty }
+    class MyDynamicBundle extends Bundle
+    class MyDynamicModule extends Module
+  }
+
+  describe("(0) Dynamic declarations") {
+    it("(0.a): name of an dynamic class of Module is anonymous") {
```

Comment:
how is this a test of anonymous? you provided the name MyDynamicModule...?

Reply:
to me a test of anonymous would be something like:

val foo = new Module { val io = IO(...)} 
assert(foo.chiselClassName = ???)